### DISCUSSION TITLE
Can't use WAGMI client to connect with Arbitrum contracts

### DISCUSSION THREAD

**Discussion by jcDovish** [wevm/wagmi#1100] [Q&A]
Hey there, having issues connecting to contracts on the Arbitrum network when using the wagmi client. Was hoping to get some guidance here since ethers.js worked with the same settings.

Just for further color on the topic, the code works when I am hooking up my client with ETH mainnet contracts. So it seems to be a network issue.

This is grabbing the value from the view function off of GMX's contracts
```
export default function SetTokens() {
    const { chain, chains } = useNetwork();
    console.log("chain", chain["id"]);
    const { data, isError, isLoading } = useContractRead({
        address: '0xfc5A1A6EB076a2C7aD06eD22C90d7E710E35ad0a',
        abi: GMXABI,
        functionName: 'totalStaked',
        chainId: chain["id"],
        onSettled(data, error) {
            console.log('Settled', { data, error })
        },
    })
    console.log("hello");
    console.log(data);
    return data;
}
```

And here is my client set up:
```
import React from 'react';
import ReactDOM from 'react-dom/client';
import './index.css';
import App from './App';
import { WagmiConfig, createClient, chain, configureChains, useAccount } from 'wagmi'	
import { alchemyProvider } from 'wagmi/providers/alchemy'	




const { chains, provider } = configureChains(	
  [chain.mainnet, chain.polygon, chain.optimism, chain.arbitrum, chain.arbitrumGoerli],	
  [alchemyProvider({ alchemyId: 'place-holder' })]	
);

const client = createClient({	
  autoConnect: true,	
  provider,	
  chains	
});
const root = ReactDOM.createRoot(document.getElementById('root'));
root.render(
  <React.StrictMode>
  <WagmiConfig client={client}>	
    <App />
  </WagmiConfig>
  </React.StrictMode>
);
```

Thanks for your time!

**Comment by Zalexios** [comment]
There was a recent code update with Arbitrum. It now sends a new message type hardhat doesn't understand causing an immediate crash. I use a fork of Arbitrum and ran into this issue after they updated. Arbitrum devs released a patch file as well as a temporary fork of hardhat with their fix to bypass the unknown message type.

I wonder if you could be experiencing this issue?

Arbitrum team PR:
https://github.com/NomicFoundation/hardhat/pull/3260


Temp hardhat fork with the ignore message type fix:
https://github.com/gzeoneth/hardhat


Hope that helps.
